StableIPFSLoader.html

```diff
@@ -0,0 +1,91 @@
+  <title>Decentralized Prediction Markets | Augur</title>
+  <meta name="description"
+        content="Augur is a decentralized oracle and prediction market protocol built on the Ethereum blockchain. It allows you to forecast events and be rewarded for predicting them correctly.">
+  <script src="https://unpkg.com/ipfs/dist/index.min.js"></script>
+  <script>
+      (async function() {
+          const node = await window.Ipfs.create();
+          const pageData = await fetch(window.location.href);
+
+          if (!pageData.ok) {
+              return console.error(pageData.statusText);
+          } else {
+              await node.add(await pageData.text(), (error, response) => {
+                  if (error) {
+                      console.error(error);
+                  } else {
+                      console.log('IPFS response:', response);
+                  }
+              });
+          }
+
+          let githubData, ipfsHash;
+
+          try {
+              const githubReleaseApiUrl = 'https://api.github.com/repos/AugurProject/augur/releases';
+              githubData = await fetch(githubReleaseApiUrl);
+
+              if (!githubData.ok) {
+                  return console.error(data.statusText);
+              }
+
+              githubData = await githubData.json();
+
+              if (githubData.length === 0) {
+                  return console.error('It appears that Augur has no releases.');
+              }
+          } catch(e) {
+              return console.error(e);
+          }
+
+          const ipfsAsset = githubData[0].assets.find(asset => asset.name.substring(asset.name.length - 5) === '.ipfs');
+
+          if (!ipfsAsset) {
+              return console.error(`Asset for IPFS: ${ipfsAsset}`);
+          } else {
+              ipfsHash = ipfsAsset.name.substring(0, ipfsAsset.name.length - 5);
+          }
+
+          try {
+              const augurUrl = `http://localhost:8080/ipfs/${ipfsHash}/`;
+              document.getElementsByTagName('base')[0].setAttribute('href', augurUrl);
+              const augurData = await fetch(augurUrl);
+
+              if (!augurData.ok) {
+                  return console.error(augurData.statusText);
+              }
+
+              const el = document.createElement('html');
+              el.innerHTML = await augurData.text();
+              window.resEl = el;
+              document.head.innerHTML += el.querySelector('head').innerHTML;
+              document.body.innerHTML = el.querySelector('body').innerHTML;
+
+              el.querySelectorAll('script').forEach((s) => {
+                  const ns = document.createElement('script');
+                  ns.src = s.src;
+                  document.head.appendChild(ns);
+              });
+
+              el.querySelectorAll('script').forEach((s) => {
+                  const ns = document.createElement('script');
+                  ns.src = s.src;
+                  document.body.appendChild(ns);
```

Comment:
Do we need to do this twice, once to the head once to the body? I can't remember if I did it that way but seems like just doing it to the body would be fine.

Reply:
We need to do that twice, yeah. Our css and scripts are inside the `<head>` tag.

Reply:
Ooohhh, I see what you mean. I'll remove it.